core/src/main/java/org/apache/calcite/sql/SqlDateTimeFormat.java

```diff
@@ -47,6 +47,9 @@
   TWENTYFOURHOURMIN("HH24MI"),
   TWENTYFOURHOURMINSEC("HH24MISS"),
   YYYYMMDDHH24MISS("YYYYMMDDHH24MISS"),
+  YYYYMMDDHH24MISSMS("YYYYMMDDHH24MISSMS"),
```

Comment:
This are not required, these should be in Raven, please connect with someone from hive team for the logic.